wagtail/admin/views/pages/edit.py

```diff
@@ -333,6 +333,15 @@ def dispatch(self, request, page_id, **kwargs):
         self.scheduled_revision = self.real_page_record.scheduled_revision
         self.page_content_type = self.real_page_record.cached_content_type
         self.page_class = self.real_page_record.specific_class
+        self.prefetch_related = self.page_class.prefetch_related
+        self.select_related = self.page_class.select_related
+        self.real_page_record = get_object_or_404(
+            self.page_class.objects.select_related(
+                *self.select_related,
+            ).prefetch_related(
+                *self.prefetch_related,
+            ),
```

Comment:
This doesn't look correct - this query will return _all_ instances of `self.page_class`, but `get_object_or_404` is only expecting one (and will thus fail with `MultipleObjectsReturned` unless only one page of that type exists).

Reply:
aside: `get_object_or_404` converts the class param to a queryset, but if you pass it a queryset, it uses that